IGNITE-14398 Documents thin client support for spring-data integration. (#8941)
apache/ignite: docs/_docs/extensions-and-integrations/spring/spring-data.adoc

== Spring Data and Apache Ignite Configuration

> Apache Ignite Spring Data integration supports connecting to Apache Ignite cluster through Apache Ignite node or
> Apache Ignite thin client. Both approaches to configuring access to Apache Ignite cluster use the same API shown below.
> Apache Ignite Spring Data integration automatically recognizes the type of provided bean and uses the appropriate cluster
> connection.

Reviewer: ```suggestion
Apache Ignite Spring Data integration supports connecting to the Apache Ignite cluster through the Apache Ignite node or Apache Ignite thin client. Both approaches to configuring access to the Apache Ignite cluster use the same API shown below. Apache Ignite Spring Data integration automatically recognizes the type of the provided bean and uses the appropriate cluster connection.
```

Reviewer: If it's not an ordered/unordered list, then we'd better not start every sentence of the paragraph from the new line.

Author: I were not intended to start every sentence of the paragraph from the new line. Just broke line after 120 characters. Сan i do that?

    }
}
----
tab:Ignite thin client connection configuration[]
[source,java]
----
@Configuration
@EnableIgniteRepositories
public class SpringAppCfg {
    /**
     * Creating Apache Ignite thin client instance bean. A bean will be passed to the IgniteRepositoryFactoryBean to
     * connect to the Ignite cluster and perform cache operations.
     */
    @Bean
    public IgniteClient igniteInstance() {
        return Ignition.startClient(new ClientConfiguration().setAddresses("127.0.0.1:10800");;
    }
}
----
--

> The configuration has to instantiate Apache Ignite bean (node) or Apache Ignite thin client bean that will be passed to
> `IgniteRepositoryFactoryBean` and will be used by all the Apache Ignite repositories in order to connect to the cluster.

Reviewer: ```suggestion
The configuration has to instantiate the Apache Ignite bean (node) or the Apache Ignite thin client bean that is passed to `IgniteRepositoryFactoryBean` and is used by all the Apache Ignite repositories in order to connect to the cluster.
```

Author: Got it. Thank you.

In the example above, the bean is initialized directly by the application and is named `igniteInstance`.
Alternatively, the following beans can be registered in your configuration and an Apache Ignite node will be started automatically:

* `IgniteConfiguration` object named as `igniteCfg` bean.
* A path to Apache Ignite's Spring XML configuration named `igniteSpringCfgPath`.

> In the case of connecting to the cluster via Apache Ignite thin client, you can alternatively register the
> `ClientConfiguration` bean named `igniteCfg` so that the Apache Ignite thin client instance will be started automatically
>  by Apache Ignite Spring Data integration.

Reviewer: ```suggestion
In the case of connecting to the cluster via Apache Ignite thin client, you can alternatively register the `ClientConfiguration` bean named `igniteCfg` so that the Apache Ignite thin client instance is started automatically by the Apache Ignite Spring Data integration.
```
